app.py

```diff
@@ -0,0 +1,224 @@
+from flask import Flask, render_template, redirect, request, session
+from hash import hash, check
+from psql import sql_select, sql_write,get_secret_key
+from service import get_card_list, get_questions, user_logged_in, fetch_data, reset_password, get_leaderboard, get_users, delete_user, log_out
+from psycopg2.extensions import AsIs 
+
+app = Flask(__name__)
+app.secret_key = get_secret_key().encode()
+
+@app.route('/')
+@app.route('/index')
+def index():
+    if user_logged_in():
+        return redirect('/home')
+    forgot = request.args.get('forgot', '')
+    stage = "1"
+    return render_template('index.html', forgot=forgot, stage=stage)
+
+@app.route('/index_action', methods=['POST'])
+def index_action():
+    forgot = request.form.get('forgot', '')
+    stage = request.form.get('stage', '')
+    email = request.form.get('email').lower()
+    answer = request.form.get('answer', '').lower()
+    query = sql_select('SELECT user_id, question, answer, username FROM users WHERE email=%s', email)
+    if query:
+        response = query[0]        
```

Comment:
nit: response is probably not the correct name for this as the `query[0]` object is actually the user. So really you probably want something like:

```
optionalUser = sql_select('SELECT user_id, question, answer, username FROM users WHERE email=%s', email)
if optionalUser:
     user = query[0]
```

Excuse my python knowledge. e.g. for typescript it would be something like:

const optionalUser: User | undefined = sql_select(`SELECT user_id, question, answer, username FROM users WHERE email=${email}`)
if (optionaluser) {
   const user = optionalUser; // we know it is User and not undefined at this point
   ....
}
```